addons/account/models/account_payment.py

```diff
@@ -502,27 +491,23 @@ def _prepare_payment_moves(self):
 
             # Compute 'name' to be used in receivable/payable line.
             rec_pay_line_name = ''
-            if payment.payment_type == 'transfer':
+            if payment.payment_is_internal:
                 rec_pay_line_name = payment.name
-            else:
-                if payment.partner_type == 'customer':
-                    if payment.payment_type == 'inbound':
-                        rec_pay_line_name += _("Customer Payment")
-                    elif payment.payment_type == 'outbound':
-                        rec_pay_line_name += _("Customer Credit Note")
-                elif payment.partner_type == 'supplier':
-                    if payment.payment_type == 'inbound':
-                        rec_pay_line_name += _("Vendor Credit Note")
-                    elif payment.payment_type == 'outbound':
-                        rec_pay_line_name += _("Vendor Payment")
-                if payment.invoice_ids:
-                    rec_pay_line_name += ': %s' % ', '.join(payment.invoice_ids.mapped('name'))
+            elif payment.partner_type == 'customer':
+                if payment.payment_type == 'inbound':
+                    rec_pay_line_name += _("Customer Payment")
+                elif payment.payment_type == 'outbound':
+                    rec_pay_line_name += _("Customer Credit Note")
+            elif payment.partner_type == 'supplier':
+                if payment.payment_type == 'inbound':
+                    rec_pay_line_name += _("Vendor Credit Note")
+                elif payment.payment_type == 'outbound':
+                    rec_pay_line_name += _("Vendor Payment")
+            if payment.invoice_ids:
+                rec_pay_line_name += ': %s' % ', '.join(payment.invoice_ids.mapped('name'))
 
             # Compute 'name' to be used in liquidity line.
-            if payment.payment_type == 'transfer':
-                liquidity_line_name = _('Transfer to %s') % payment.destination_journal_id.name
-            else:
-                liquidity_line_name = payment.name
```

Comment:
outbound + internal transfer = `Transfer from <journal>`
inbound + internal transfer = `Transfer to <journal>`